packages/panzer/dof-mgr/src/Panzer_IntrepidFieldPattern.cpp

```diff
@@ -257,7 +263,13 @@ namespace panzer {
 
     if(numCells>0) {
       Intrepid2::CellTools<PHX::Device> cellTools;
-      cellTools.mapToPhysicalFrame(coords,localCoords,cellVertices,intrepidBasis_->getBaseCellTopology());
+      // For backwards compatability, allow the FE basis to supply the mesh cell topology (via the FEM base cell topo)
+      // If provided, use the mesh topology directly
+      if (meshCellTopology==Teuchos::null) {
+        cellTools.mapToPhysicalFrame(coords,localCoords,cellNodes,intrepidBasis_->getBaseCellTopology());
```

Comment:
Why don't you pass directly the (HGRAD) basis to the `mapToPhysicalFrame` function?
https://github.com/trilinos/Trilinos/blob/master/packages/intrepid2/src/Cell/Intrepid2_CellTools.hpp#L1021-L1024

In fact, when you pass the topology, intrepid2 has to build the corresponding HGRAD basis.

Reply:
Good question. That particular bit was there from before my time, but we should examine. Would you get different behavior passing in a higher Intrepid basis for any reason?

Reply:
Anyway, my thought process was ::

1) If the user has a first-order mesh, then cellNodes will just be cellVertices and to be compatible with computing the map, you'd want a first-order basis in that calculation.
2) If the user has a higher-order mesh, the cellNodes is larger and corresponds with the order of the mesh.

Essentially, in either case, we are using a basis which corresponds to the mesh order to create this map.
If no mesh topology is supplied, we are using the intrepidbasis to determine the mesh topology (which will be the lowest-order topology).

Reply:
I see your reasoning. What is the `intrepidBasis`? Is it the basis used for mapping between the ref element and the physical element (e.g., for affine meshes it would be HGRAD_C1, for quadratic meshes it would be HGRAD_C2)? If so, it's better to directly pass the basis, which will also work for higher order (>2) meshes.

Reply:
In this case, `intrepidBasis_` is the FE basis, which at least theoretically could be any order and, generally, does not match the mesh order. As this snippet indicates, we are not explicitly constructing or carrying around an additional intrepid basis specifically for the mapping -- but rather constructing it within `mapToPhysicalFrame`.

If you see this as problematic, I can look into constructing a basis upstream specifically for this purpose.
I guess it may depend on if this function is called many time and a temporary basis keeps getting made over and over.
There may also be some utility to eliminating this if statement and always explicitly requiring the mesh topology (or a "mesh-order" intrepid basis, like you point out).